### PR TITLE
Add rules for agent

### PR DESCRIPTION
**Pull request by qiujian16**
1. manifestwork rules so work agent can get work
2. events rules for event recorder

### CODE REVIEW

File: pkg/hub/spokecluster/manifests/spokecluster-role.yaml [hunk 1]
```diff
@@ -3,5 +3,16 @@ kind: Role
 metadata:
   name: {{ .SpokeClusterName }}
```
Comment (deads2k) [inline pkg/hub/spokecluster/manifests/spokecluster-role.yaml]: I think we're going to need names that distinguish
1. spoke-operator
2. spoke-registration
3. spoke-work

I'm having trouble keeping track and we don't have that many yet

Comment (pmorie) [reply pkg/hub/spokecluster/manifests/spokecluster-role.yaml]: Agree.

I'm also not sure if this is the right place for these rules to live, but I think we can go with it for now.

Comment (qiujian16) [reply pkg/hub/spokecluster/manifests/spokecluster-role.yaml]: i will change the name of these roles so it would be easier to distinguish

**Pull request merged**